scripts/download.sh

```diff
@@ -35,5 +35,8 @@ else
 fi
 
 if [ ! -z "$STATS_IP" ]; then
+  while ! echo exit | nc -u -w1 $NODE_IP $STATS_PORT;
+  do sleep 120;
+  done
```

Comment:
There's virtually no cost to polling consistently, so I'd do every 1 second, and maybe log conditionally every time 15 iterations go by. I would also timeout after 5 minutes. Interestingly, the ldb file will exist, so the apps the need it can startup and make progress, but the reflector itself will be blocked.

I thought about making this another cli app or mode the reflect runs in. It wouldn't be too much work. My gut feeling is just to try this, and if it proves cumbersome and awkward, we should implement this script in go.

Reply:
>  Interestingly, the ldb file will exist, so the apps the need it can startup and make progress, but the reflector itself will be blocked.

The current defaults for the `ldb-waiter` suck for this case actually.  It's 1 minute by default, which means if the LDB file hasn't been caught up to latest by the reflector actually running and freshening up the LDB, then the apps won't be able to continue unless they bump their "allowed lag" (misnamed as [*minLatency* in the configuration](https://github.com/segmentio/k2/blob/6b042683b4f6ecec1786c95700f487b5ea294493/pkg/apis/v1beta1/segmentapplication/app_container.go#L322-L329)).

I wanna propose bumping the default **allowed** lag up a lot, to say 1 hour or more.

Reply:
oh interesting, TIL

Reply:
> I wanna propose bumping the default allowed lag up a lot, to say 1 hour or more.
I guess this would involve some code/config changes in k2?

Reply:
@zhou-hongyu : yup!  `k2` is what sets the default, the `ldb-waiter` code is a slim wrapper on top of ctlstore's `GetLedgerLatency()` method, it doesn't even have defaults for the config parameters.

* https://github.com/segmentio/images/blob/199d67473930004b18daba2902d63b1779bec5b8/ldb-waiter/main.go#L34